page-objects/src/components/sidebar/WelcomeContent.ts

```diff
@@ -0,0 +1,40 @@
+import { By, WebElement } from "selenium-webdriver";
+import { ViewSection } from "./ViewSection";
+import { AbstractElement } from "../AbstractElement";
+
+export class WelcomeContentButton extends AbstractElement {
+  constructor(
+    private panel: WebElement,
+    public welcomeSection: WelcomeContentSection
```

Comment:
You don't really need to declare the arguments as fields here.
The first (`panel`) argument is the main reference for selenium, after the object is initialized, `this` is populated with whatever element is found by this reference.
The second (`welcomeSection`) argument is accessible as a protected field `enclosingElement`, or in `getEnclosingElement()` method publicly.